db/dcrpg/queries.go

```diff
@@ -4096,3 +4165,306 @@ func UpdateBlockNextByNextHash(db SqlExecutor, currentNext, newNext string) erro
+	N := ind1 - ind0 + 1
+	if N == 0 {
+		return []int32{}, nil
+	}
+	if N < 0 {
+		return nil, fmt.Errorf("Cannot retrieve block size range (%d>%d)",
+			ind0, ind1)
+	}
+
+	blockSizes := make([]int32, 0, N)
+
+	stmt, err := db.Prepare(internal.SelectBlockSizeRange)
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.QueryContext(ctx, ind0, ind1)
+	if err != nil {
+		log.Errorf("Query failed: %v", err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var blockSize int32
+		if err = rows.Scan(&blockSize); err != nil {
+			log.Errorf("Unable to scan for block size field: %v", err)
+		}
+		blockSizes = append(blockSizes, blockSize)
+	}
+	if err = rows.Err(); err != nil {
+		log.Error(err)
+	}
+
+	return blockSizes, nil
+}
+
+// RetrieveSDiff returns the stake difficulty for block at the specified chain
+// height.
+func RetrieveSDiff(ctx context.Context, db *sql.DB, ind int64) (float64, error) {
+	var sbits int64
+	err := db.QueryRowContext(ctx, internal.SelectSBitsByHeight, ind).Scan(&sbits)
+	return dcrutil.Amount(sbits).ToCoin(), err
+}
+
+// RetrieveSDiffRange returns an array of stake difficulties for block range
+// ind0 to ind1.
+func RetrieveSDiffRange(ctx context.Context, db *sql.DB, ind0, ind1 int64) ([]float64, error) {
+	N := ind1 - ind0 + 1
+	if N == 0 {
+		return []float64{}, nil
+	}
+	if N < 0 {
+		return nil, fmt.Errorf("Cannot retrieve sdiff range (%d>%d)",
+			ind0, ind1)
+	}
+	sdiffs := make([]float64, 0, N)
+
+	stmt, err := db.Prepare(internal.SelectSBitsRange)
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.QueryContext(ctx, ind0, ind1)
+	if err != nil {
+		log.Errorf("Query failed: %v", err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var sbits int64
+		if err = rows.Scan(&sbits); err != nil {
+			log.Errorf("Unable to scan for sdiff fields: %v", err)
+		}
+		sdiffs = append(sdiffs, dcrutil.Amount(sbits).ToCoin())
+	}
+	if err = rows.Err(); err != nil {
+		log.Error(err)
+	}
+
+	return sdiffs, nil
+}
+
+// RetrieveLatestBlockSummary returns the block summary for the best block.
+func RetrieveLatestBlockSummary(ctx context.Context, db *sql.DB) (*apitypes.BlockDataBasic, error) {
+	bd := apitypes.NewBlockDataBasic()
+
+	var winners []string
+	var timestamp dbtypes.TimeDef
+	var isValid bool
+	var val, sbits int64
+	err := db.QueryRowContext(ctx, internal.SelectBlockDataBest).Scan(
+		&bd.Hash, &bd.Height, &bd.Size, &bd.Difficulty, &sbits, &timestamp,
+		&bd.PoolInfo.Size, &val, pq.Array(&winners), &isValid)
+	if err != nil {
+		return nil, err
+	}
+	bd.PoolInfo.Value = dcrutil.Amount(val).ToCoin()
+	bd.PoolInfo.ValAvg = bd.PoolInfo.Value / float64(bd.PoolInfo.Size)
+	bd.Time = apitypes.TimeAPI{S: timestamp}
+	bd.PoolInfo.Winners = winners
+	bd.StakeDiff = dcrutil.Amount(sbits).ToCoin()
+	return bd, nil
+}
+
+// RetrieveDiff returns the difficulty in the last 24hrs or immediately after
+// 24hrs.
```

Comment:
This comment is equally confusing as the other one.  I know this was already like this, but if you can think of a better explanation, I figure it's a good time to fix it.